src/test/java/com/microsoft/sqlserver/jdbc/unit/lobs/LobsStreamingTest.java

```diff
@@ -0,0 +1,313 @@
+package com.microsoft.sqlserver.jdbc.unit.lobs;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.sql.Clob;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.NClob;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.IntStream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Utils;
+import com.microsoft.sqlserver.testframework.util.RandomUtil;
+
+
+@RunWith(JUnitPlatform.class)
+public class LobsStreamingTest extends AbstractTest {
+
+    private static final int LOB_ARRAY_SIZE = 250; // number of rows to insert into the table and compare
+    private static final int LOB_LENGTH_MIN = 8000;
+    private static final int LOB_LENGTH_MAX = 32000;
+
+    static String tableName;
+    static String escapedTableName;
+
+    @BeforeEach
+    public void init() throws SQLException {
+        tableName = RandomUtil.getIdentifier("streamingTest");
+        escapedTableName = AbstractSQLGenerator.escapeIdentifier(tableName);
+    }
+
+    private String getRandomString(int length) {
+        String validCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890!@#$%^&*()-=_+,./;'[]<>?:{}|`~\"\\";
+        StringBuilder salt = new StringBuilder();
+        Random rnd = new Random();
+        while (salt.length() < length) {
+            int index = (int) (rnd.nextFloat() * validCharacters.length());
+            salt.append(validCharacters.charAt(index));
+        }
+        String saltStr = salt.toString();
+        return saltStr;
+
+    }
+
+    private String getStringFromInputStream(InputStream is, Charset c) {

```

Comment:
Close the Scanner here.

Reply:
Closing scanner closes the Inputstream. The stream is needed at a later part of the test.